### ISSUE TITLE
systemd-journald coredumps do not show up in coredumpctl

### ISSUE THREAD

**Issue by strugee** [systemd/systemd#15688]
**systemd version the issue has been seen with**

241 (I looked into this being fixed in later versions though, both in `NEWS` and in the git log for `src/coredump/`)

**Used distribution**
 
Debian 10

**Expected behaviour you didn't see**

 `coredumpctl list` shows a systemd-journald coredump.

**Unexpected behaviour you saw**
 
`systemd-coredumpd` was apparently unable to handle this situation. Kernel logs indicated that `systemd-journald` had dumped core, and `systemd-coredumpd` had handled it (unfortunately I didn't think to capture `dmesg` logs before I rebooted, otherwise I'd post them). Sure enough:

```
% ls /var/lib/systemd/coredump                                                                                   
core.systemd-journal.0.3b161c24391f4ab5ab11422941817b66.41986.1588514632000000.lz4
```
 and yet:

```
% sudo coredumpctl
No coredumps found.
```

**Steps to reproduce the problem**

1. Install and enable `systemd-coredumpd`
2. Cause the journal to dump core for some reason (`killall -ABRT /lib/systemd/systemd-journald` works just fine to reproduce, although it apparently won't log in `dmesg`)
3. `sudo coredumpctl list`

Note that if I send SIGABRT to a random bash process, that gets dumped and logged just fine, so the daemon is definitely working.

**Comment by poettering** [comment]
So the problem is that we use the journal to track coredumps, but if it's the journal that coredumps we can't really use the journal to store this case, and hence special case it.

**Comment by strugee** [comment]
Right, I figured that was the case. To me though a reasonable behavior would be to wait until the journal is back up and then flush that information (either by just blocking, via another scheduled job, or maybe `coredumpctl` could just figure out that that information was missing at runtime and put it in the journal)
